src/QMCWaveFunctions/PlaneWave/PWOrbitalSetT.cpp

```diff
@@ -0,0 +1,146 @@
+    OHMMS::Controller->abort();
+  }
+  //drop G points for the given TwistAngle
+  const std::vector<int>& inputmap(myBasisSet->inputmap);
+  for (int ig = 0; ig < ng; ig++)
+  {
+    if (inputmap[ig] > -1)
+      (*(this->C))[jorb][inputmap[ig]] = coefs[ig];
+  }
+}
+
+template<class T>
+void PWOrbitalSetT<T>::addVector(const std::vector<RealType>& coefs, int jorb)
+{
+  int ng = myBasisSet->inputmap.size();
+  if (ng != coefs.size())
+  {
+    app_error() << "  Input G map does not match the basis size of wave functions " << std::endl;
+    OHMMS::Controller->abort();
+  }
+  //drop G points for the given TwistAngle
+  const std::vector<int>& inputmap(myBasisSet->inputmap);
+  for (int ig = 0; ig < ng; ig++)
+  {
+    if (inputmap[ig] > -1)
+      (*(this->C))[jorb][inputmap[ig]] = coefs[ig];
+  }
+}
+
+template<class T>
+void PWOrbitalSetT<T>::evaluateValue(const ParticleSet& P, int iat, ValueVector& psi)
+{
+  //Evaluate every orbital for particle iat.
+  //Evaluate the basis-set at these coordinates:
+  //myBasisSet->evaluate(P,iat);
+  myBasisSet->evaluate(P.activeR(iat));
+  MatrixOperators::product<T>(*(this->C), myBasisSet->Zv, psi);
+}
+
+template<class T>
+void PWOrbitalSetT<T>::evaluateVGL(const ParticleSet& P, int iat, ValueVector& psi, GradVector& dpsi, ValueVector& d2psi)
+{
+  //Evaluate the orbitals and derivatives for particle iat only.
+  myBasisSet->evaluateAll(P, iat);
+  MatrixOperators::product<T>(*(this->C), myBasisSet->Z, this->Temp);
+  const T* restrict tptr = this->Temp.data();
+  for (int j = 0; j < this->OrbitalSetSize; j++, tptr += PW_MAXINDEX)
+  {
+    psi[j]   = tptr[PW_VALUE];
+    d2psi[j] = tptr[PW_LAP];
+    dpsi[j]  = GradType(tptr[PW_GRADX], tptr[PW_GRADY], tptr[PW_GRADZ]);
+  }
+}
+
+template<class T>
+void PWOrbitalSetT<T>::evaluate_notranspose(const ParticleSet& P,
+                                        int first,
+                                        int last,
+                                        ValueMatrix& logdet,
+                                        GradMatrix& dlogdet,
+                                        ValueMatrix& d2logdet)
+{
+  for (int iat = first, i = 0; iat < last; iat++, i++)
+  {
+    myBasisSet->evaluateAll(P, iat);
+    MatrixOperators::product<T>(*(this->C), myBasisSet->Z, this->Temp);
+    const T* restrict tptr = this->Temp.data();
+    for (int j = 0; j < this->OrbitalSetSize; j++, tptr += PW_MAXINDEX)
+    {
+      logdet(i, j)   = tptr[PW_VALUE];
+      d2logdet(i, j) = tptr[PW_LAP];
+      dlogdet(i, j)  = GradType(tptr[PW_GRADX], tptr[PW_GRADY], tptr[PW_GRADZ]);
+    }
+  }
+}
+
+// Class concrete types from T
+// NOTE: This class only gets compiled if QMC_COMPLEX is defined, thus it is inherently complex
+// template class PWOrbitalSetT<double>;
+// template class PWOrbitalSetT<float>;
+template class PWOrbitalSetT<std::complex<double>>;
+// NOTE: MatrixOperators::product does not support floats
```

Comment:
MatrixOperators::product requires A, B and C to be the same type, but that type can be double, std::complex<double>, float, or std::complex<float>.